snet_cli/mpe_service_command.py

```diff
@@ -171,6 +174,36 @@ def print_service_metadata_from_registry(self):
         metadata      = self._get_service_metadata_from_registry()
         self._printout(metadata.get_json_pretty())
 
+    def _service_status(self, url, secure=True):
+        try:
+            channel = open_grpc_channel(endpoint=url)
+            stub = heartb_pb2_grpc.HealthStub(channel)
+            response = stub.Check(heartb_pb2.HealthCheckRequest(service=""), timeout=10)
+            if response!=None and response.status == 1:
+                return True
+            return False
+        except Exception as e:
+            return False
+
+    def print_service_status(self):
+        metadata = self._get_service_metadata_from_registry()
+        if self.args.group_name != None:
+            groups = {self.args.group_name: metadata.get_endpoints_for_group(self.args.group_name)}
+        else:
+            groups = metadata.get_all_endpoints_with_group_name()
+        srvc_status = {}
+        for grp in groups:
+            srvc_status[grp] = {}
+            if groups[grp] == []:
+               srvc_status.pop(grp)
+            for endpoint in groups[grp]:
+                status = "Available" if self._service_status(url=endpoint)  else "Not Available"
+                srvc_status[grp].update({"endpoint": endpoint, "status": status})
```

Comment:
update? It will not work for several endpoints because you will print information only about the last endpoint.

I would propose to rewrite like this (which also simplify code):
```python
        srvc_status = defaultdict(list)
        for grp in groups:
            for endpoint in groups[grp]:
                status = "Available" if self._service_status(url=endpoint)  else "Not Available"
                srvc_status[grp].append({"endpoint": endpoint, "status": status})
        if srvc_status == {}:
            self._printout("Error: No endpoints found to check service status.")
            return
        self._pprint(srvc_status)

```